core/ledger/kvledger/txmgmt/privacyenabledstate/snapshot.go

```diff
@@ -483,5 +502,5 @@ func (c *cursor) currentNamespace() string {
 }
 
 type SnapshotPvtdataHashesConsumer interface {
-	ConsumeSnapshotData(namespace, coll string, keyHash []byte, version *version.Height) error
+	ConsumeSnapshotData(namespace, coll string, keyHash []byte, valueHash []byte, version *version.Height) error
```

Comment:
nit. 

Rather than passing 5 parameters, we could define a `PvtStateHash` struct in the `ledger_interface.g`o and use it here, `purgeMgr`, and `pvtdatastore` packages.

Reply:
I too felt that and considered alternatives before choosing this one. Specifically, about your suggestion, we have kept `ledger_interface.go` to define types that are exposed to other components not for the cross components within ledger. For components within ledger, it's better to let them have the explicit dependency if they need. I do not find this worth either breaking the pattern for `ledger_interface.go` or introducing a dependency.

Reply:
I remember having this discussion 2 years ago. We decided to have a place within ledger to share common types (within ledger). Finally, we ended up using the `ledger_interface.go` (i think, something related to pvtdata) deferring the former approach.